General queries/Baseline Comparison.txt

```diff
@@ -0,0 +1,239 @@
+    | summarize makeset(MachineId)
+};
+// Function to get network communications given a list of computer names and how far back to look
+let GetNetworkEvents=(InMachineId: dynamic, LeftEventTime: datetime) {
+    NetworkCommunicationEvents
+    | where "Network Communication" in (ReturnSets)
+    | where EventTime > LeftEventTime
+    | where MachineId in~ (InMachineId)
+    | summarize EventTime=max(EventTime), count() by RemoteUrl, MachineId
+    | extend UrlSplit=split(RemoteUrl, ".") // Split the levels of the URL
+    // If there is only one level (for an internal communication that uses your DNS search suffix), then only use that level
+    // Otherwise combine the top two levels and use those as the URLRoot
+    | extend UrlRoot=iff(UrlSplit[-2] == "", UrlSplit[0], strcat(tostring(UrlSplit[-2]), ".", tostring(UrlSplit[-1])))
+    | summarize EventTime=max(EventTime), Count=sum(count_), AdditionalData=makeset(RemoteUrl) by UrlRoot, MachineId
+    | project EventTime, Entity=UrlRoot, Count, AdditionalData=tostring(AdditionalData), MachineId, DataType="Network Communication"
+};
+// Function to get process creates given a list of computer names and how far back to look
+let GetProcessCreates=(InMachineId: dynamic, LeftEventTime: datetime) {
+    ProcessCreationEvents
+    | where "Process Creation" in (ReturnSets)
+    | where EventTime > LeftEventTime
+    | where MachineId in~ (InMachineId)
+    // Replace known path for mpam files as they are dynamically named and likely to be unique on each machine
+    | extend FileName=iff(FolderPath matches regex @"([A-Z]:\\Windows\\ServiceProfiles\\NetworkService\\AppData\\Local\\Temp\\mpam-)[a-z0-9]{7,8}\.exe", "mpam-RANDOM.exe", FileName)
+    // Replace known path for AM delta patch files as they jump frequently and not likely to be exact on each machine
+    | extend FileName=iff(FolderPath matches regex @"([A-Z]:\\Windows\\SoftwareDistribution\\Download\\Install\\AM_Delta_Patch_)[0-9\.]+\.exe", "AM_Delta_Patch_Version.exe", FileName)
+    | summarize EventTime=max(EventTime), Count=count(), AdditionalData=makeset(FolderPath) by FileName, MachineId
+    // Replace various mbam executables that are semiunique-generated with some text to help reduce noise
+    | project EventTime, Entity=FileName, Count, AdditionalData=tostring(AdditionalData), MachineId, DataType="Process Creation"
+};
+// Function to get powershell commands given a list of computer names and how far back to look
+let GetPSCommands=(InMachineId: dynamic, LeftEventTime: datetime) {
+    MiscEvents
+    | where "PowerShell Command" in (ReturnSets)
+    | where EventTime > LeftEventTime
+    | where MachineId in~ (InMachineId)
+    | where ActionType == 'PowerShellCommand'
+    // Remove two different signatures for scripts being executed which cause a lot of noise
+    // The first signature matches scripts generated as part of testing execution policy
+    // The second signature matches scripts generated by SCCM
```

Comment:
maybe could add filtering on the initiatingprocessparentname, on initiating process commandline or change the regex to check also for the .ps1 suffix.

Reply:
Added in latest commits